### ISSUE TITLE
vec3f 16 byte alignment vs 12 byte size

### ISSUE THREAD

**Issue by mikeskk** [gpuweb/gpuweb#4192]
The fact that vec3f, despite having a size of 12 bytes, has an alignment of 16, should be explicitly mentioned in the docs. 

This is extremely confusing to beginners, such as myself, and this doesn't seem to be mentioned almost anywhere on the internet. Aside from empirical evidence, this is the only source I've found on the topic: https://sotrh.github.io/learn-wgpu/showcase/alignment/ .

**Comment by shannon** [comment]
@mikeskk I agree this is one of the trickier things I ran into when learning the API. It is documented, it is just under WGSL

https://www.w3.org/TR/WGSL/#memory-layouts

**Comment by greggman** [comment]
You might find [this](https://webgpufundamentals.org/webgpu/lessons/resources/wgsl-offset-computer.html#x=5d000001005000000000000000003d88886237289d13c4320e1a9ba574a728071755edcf12a0ea5e1a4ed7aa1e87736861e35c2f844b34a16006299e7d6adc8359c9a53faa4066e168d49dcb7d4918db8279c973cbc72f5b0fdadedf5fe153c8ff4df50000) helpful

**Comment by FL33TW00D** [comment]
Yeah this completely ruined my week a while back. It's totally documented and clear, but so unintuitive.

**Comment by dneto0** [comment]
Another thing that is a gotcha is that the spec usually talks about vector types in the fully-elaborated template spelling, e.g. `vec3<f32>` rather than the shorter alias `vec3f`.   Always look up rules about vector types by looking for the fully-elaborated template spelling.

**Comment by kainino0x** [comment]
It would be nice to document this somewhere that people will notice more readily. It should maybe mention "vec3f"/"vec3i"/"vec3u" by name for searchability. But I don't know where it should go that would actually be easier to find. Are there any suggestions?
